docs/tutorials/typescript/workqueue-slackbot/code/bot/modules/workqueue.ts

```diff
@@ -0,0 +1,390 @@
+  try {
+    // Use the WebClient to fetch channel info
+    const channelInfo = await client.conversations.info({channel: channelId});
+    if (channelInfo.ok) {
+      return channelInfo.channel?.name ?? "Unknown Channel";
+    } else {
+      await say("Unable to fetch channel info.");
+      return "Unknown Channel";
+    }
+  } catch (error) {
+    console.error(error);
+    await say("An error occurred while fetching channel info.");
+    return "Unknown Channel";
+  }
+}
+
+async function checkIfWorkqueueExists(workflowId: string) {
+  try {
+    const describeResult =
+      await temporalClient.workflowService.describeWorkflowExecution({
+        namespace: process.env.TEMPORAL_CLOUD_NAMESPACE!,
+        execution: {
+          workflowId: workflowId,
+        },
+      });
+    // Check if the Workflow is currently running
+    const status = describeResult.workflowExecutionInfo?.status;
+    // 1 corresponds to RUNNING
+    if (status === 1) {
+      console.log(`Workflow with ID ${workflowId} is currently running.`);
+      return true;
+    } else {
+      console.log(`Workflow with ID ${workflowId} is not running.`);
+      return false;
+    }
+  } catch (error: any) {
+    if (error.message.includes("NOT_FOUND")) {
+      console.log(`No workflow found with ID ${workflowId}.`);
+      return false;
+    }
+    console.error("Error describing workflow execution:", error);
+    throw error;
+  }
+}
+
+async function createNewWorkqueue(workflowid: string): Promise<void> {
+  await temporalClient.workflow.start("workqueue", {
+    taskQueue: `${process.env.IQ_ENV}-temporal-iq-task-queue`,
+    workflowId: workflowid,
+  });
+}
+
+async function queryWorkqueue(
+  workflowId: string,
+  say: SayFn
+): Promise<WorkqueueData[]> {
+  try {
+    const handle = temporalClient.workflow.getHandle(workflowId);
+    const result = await handle.query<WorkqueueData[]>("getWorkqueueData");
+    console.log("Current workqueue data:", result);
+    return result;
+  } catch (error) {
+    console.error("Error querying workqueue data:", error);
+    await say("An error occurred while Querying the Work Queue.");
+    return [];
+  }
+}
+
+async function signalAddWork(params: WorkqueueData, say: SayFn): Promise<void> {
+  try {
+    await temporalClient.workflow.signalWithStart("workqueue", {
+      workflowId: params.channelName,
+      taskQueue: `${process.env.IQ_ENV}-temporal-iq-task-queue`,
+      signal: "addWorkqueueData",
+      signalArgs: [params],
+    });
+  } catch (error) {
+    console.error("Error signaling workqueue data:", error);
+    await say("An error occurred while Signaling the Work Queue.");
+  }
+}
+
+export async function signalClaimWork(
+  channelName: string,
+  workId: string,
+  claimantId: string,
+  message: GenericMessageEvent,
+  userId: string,
+  say: SayFn
+) {
+  try {
+    const handle = temporalClient.workflow.getHandle(channelName);
+    await handle.signal("claimWork", {workId, claimantId});
```

Comment:
Use the `claimWorkSignal` object here instead of a bare string. That will give you the benefits of type safety. Same for other signals/queries.